website/docs/articles/publishing-terraform-modules.md

```diff
@@ -0,0 +1,142 @@
+---
+sidebar_label: Publishing Terraform modules to the Registry
+---
+
+
+# Publishing Terraform modules to the Registry
+
+This guide walks through the process of creating, developing, and publishing Terraform modules to the Terraform Registry.
+
+The Terraform Registry serves as a central repository where organizations can publish and share their Terraform modules with the community. It provides a standardized way to discover, distribute, and version infrastructure as code components. Our organization maintains a collection of modules in the PagoPA namespace, which you can find [here](https://registry.terraform.io/namespaces/pagopa).
+
+We'll cover the entire lifecycle from initialization to publication, including our versioning system and automated workflows.
+
+Our Terraform modules are organized in the `infra/modules` directory, with each module containing its own code, tests, and examples.
+
+## Creating a New Module
+
+### Initialize the module
+
+To create a new module, we provide an automated initialization script called `add-module.sh` located in the `infra/scripts` directory. This script handles all the necessary setup steps and ensures consistency across module creation.
+
+The script accepts the following parameters:
+
+```bash
+./add-module.sh --name <module_name> [--gh-org <github_organization>] [--provider <provider_name>]
+```
+
+Parameters explained:
+- `--name`: Required. The name of your module (e.g., `azure_api_management`)
+- `--gh-org`: Optional. The GitHub organization where the module's sub-repository will be created
+- `--provider`: Optional. Defaults to `azurerm`. Specifies the cloud provider (e.g., `aws`, `azurerm`)
+
+### What the Initialization Script Does
+
+The script performs several important steps:
+
+1. Validates if the module directory already exists to prevent accidental overwrites
+2. Creates the module directory structure under `infra-modules/<provider_name>/`
+3. Generates a `package.json` file for version management
+4. Creates a dedicated sub-repository with Terraform Registry compliant naming
+5. Initializes the repository with the module's base code
+
+After successful initialization, you'll need to:
+1. Contact DevEx team members to add the new repository to:
+   - The dx-pagopa-bot PAT
+   - Track it in the eng-github-authorization
+
+## Module Development and Versioning
+
+We use [Changeset](https://github.com/changesets/changesets) for version management and changelog generation. This helps maintain a clear history of changes and ensures proper semantic versioning.
+
+### Understanding Semantic Versioning
+
+Semantic Versioning (SemVer) uses three numbers (MAJOR.MINOR.PATCH) to indicate the nature of changes:
```

Comment:
I'd remove this paragraph, just say 'modules must use semantic versioning' and link some external resource.